lib/types/keypair.ex

```diff
@@ -49,4 +61,13 @@ defmodule Kadena.Types.KeyPair do
   defp validate_caps_list({_arg, nil}), do: {:ok, nil}
   defp validate_caps_list({_arg, [%Cap{} | _tail] = clist}), do: {:ok, clist}
   defp validate_caps_list({arg, _clist}), do: {:error, [{arg, :invalid}]}
+
+  defp create_clist({arg, caps}) when is_list(caps) do
+    caps
+    |> Enum.map(fn x -> Enum.map(x, fn {key, value} -> {String.to_atom(key), value} end) end)
+    |> Enum.map(fn cap -> Cap.new(cap) end)
+    |> (&validate_caps_list({arg, &1})).()
+  end
```

Comment:
`String.to_atom/1` creates atoms dynamically and atoms are not garbage-collected. Please review if you can avoid the creation of atoms since it is not recommended for user-supplied inputs.

I think the caps list only requires `name` and `args`, so we could easily avoid creating atoms